raiden/network/proxies/user_deposit.py

```diff
@@ -23,14 +48,113 @@ def __init__(
             CONTRACT_USER_DEPOSIT,
         )
 
+        self.client = jsonrpc_client
+
         self.address = user_deposit_address
+        self.node_address = self.client.address
         self.contract_manager = contract_manager
+
         self.proxy = jsonrpc_client.new_contract_proxy(
             self.contract_manager.get_contract_abi(CONTRACT_USER_DEPOSIT),
             to_normalized_address(user_deposit_address),
         )
 
-        self.client = jsonrpc_client
+        self.deposit_lock = RLock()
+
+    def token_address(
+            self,
+            block_identifier: BlockSpecification,
+    ) -> Address:
+        return to_canonical_address(self.proxy.contract.functions.token().call(
+            block_identifier=block_identifier,
+        ))
+
+    def get_total_deposit(
+            self,
+            address: Address,
+            block_identifier: BlockSpecification,
+    ):
+        fn = getattr(self.proxy.contract.functions, 'balances')
```

Comment:
Why do this here, but not in other places?

Reply:
You mean `getattr`? if so, i've changed this.

Reply:
Exactly, just to be consistent with the lines above.